### PR TITLE
[Validator] deprecate non-string constraint violation codes

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/32066#discussion_r294106130
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Component/Validator/Tests/Violation/ConstraintViolationBuilderTest.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Violation;
+
+use Symfony\Bundle\TwigBundle\Tests\TestCase;
```
Comment (Tobion) [inline src/Symfony/Component/Validator/Tests/Violation/ConstraintViolationBuilderTest.php]: wrong TestCase class

Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Violation/ConstraintViolationBuilderTest.php]: already fixed 😄

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thinking about it, would it be feasible to use `int` instead of `string`. That seems more "natural" in PHP to have error codes as `int`s.

**Comment by xabbuh** [comment]
@fabpot All our built-in constraints already use UUID strings as the code.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**